src/device/test/GPU_Device.cmake

```diff
@@ -11,37 +11,42 @@
 
 set( test_sources
   gpu_hello_rt_api.cu
-  # The low level Driver-API is not really supported by NVidia anymore.
-  # gpu_hello_driver_api.cc
-  )
+  gpu_device_info.cu
+  gpu_dual_call_test.cu
+)
 
-# set( cuda_sources
-#   gpu_kernel.cu
-#   vector_add.cu
-#   )
+set(cuda_headers
+  basic_kernels.hh
+  Dual_Call.hh
+)
+
+set(cuda_sources
+  basic_kernels.cu
+  Dual_Call.cu
+)
 
 # ---------------------------------------------------------------------------- #
 # Build Unit tests
 # ---------------------------------------------------------------------------- #
 
 # Stuff cuda code into a test library.
-# add_library( Lib_device_test SHARED ${cuda_sources} )
-
-#add_executable( Ut_gpu_hello_driver_api_exe gpu_hello_driver_api.cc )
-#target_link_libraries( Ut_gpu_hello_driver_api_exe Lib_device )
-
-add_executable( Ut_gpu_hello_rt_api_exe gpu_hello_rt_api.cu )
-target_link_libraries( Ut_gpu_hello_rt_api_exe Lib_dsxx )
-# target_link_libraries( Ut_gpu_hello_rt_api_exe Lib_device_test Lib_dsxx )
-#target_link_libraries( Ut_gpu_hello_rt_api_exe Lib_device )
+add_component_library(
+   TARGET       Lib_device_test
+   TARGET_DEPS  Lib_dsxx Lib_device
+   LIBRARY_NAME device_test
+   LIBRARY_TYPE STATIC
+   SOURCES      "${cuda_sources}"
+   HEADERS      "${cuda_headers}" )
+set_property(TARGET Lib_device_test PROPERTY CUDA_SEPARABLE_COMPILATION ON)
 
```

Comment:
Awesome sauce!

* Is STATIC required? If so, can we make this automatic for `*.cu` files?
* Do you think we can move the `CUDA_SEPARABLE_COMPILATION` into `add_component_library`, maybe enable if any source file matches `*.cu`?

Reply:
* Yes, static is required as far as I can tell. I found that in this presentation (http://on-demand.gputechconf.com/gtc-express/2012/presentations/gpu-object-linking.pdf). Yes, automatically doing this for .cu files seems like a great idea!
* I think we can do this through the same mechanism as the automatic static build.
 
Do you want me to add those features as part of this PR?

Reply:
I'll add a redmine ticket to track this.